src/pages/TabHome.tsx

```diff
@@ -83,7 +83,9 @@ function useDaysBeforePeriod(): DaysBeforePeriod {
 
   const dateOfFinish = new Date(startDate);
   dateOfFinish.setDate(dateOfFinish.getDate() + Number(cycleLength));
+  dateOfFinish.setHours(0, 0, 0, 0);
```

Comment:
As I can see we don't use hours/minutes/seconds in any place

Maybe just call `setHours` once when we save data to the storage/Context and remove all unnecessary `setHours`?